src/System.Management.Automation/engine/regex.cs

```diff
@@ -86,15 +86,17 @@ internal string PatternConvertedToRegex
         /// </summary>
         /// <param name="pattern">The wildcard pattern to match.</param>
         /// <returns>The constructed WildcardPattern object.</returns>
-        /// <remarks> if wildCardType == None, the pattern does not have wild cards</remarks>
+        /// <remarks>If wildCardType == None, the pattern does not have wild cards.</remarks>
         public WildcardPattern(string pattern)
         {
             if (pattern == null)
             {
-                throw PSTraceSource.NewArgumentNullException("pattern");
+                throw PSTraceSource.NewArgumentNullException(nameof(pattern));
             }
 
             Pattern = pattern;
+
+            Init();
```

Comment:
I'm not sure about this. There are places where a `WildcardPattern` is created but it's not used because the string doesn't contains wildcard characters. The `GetEventCommands.cs` you touched here has examples like this.
Are there any evidence this is better than the lazy init?

Reply:
> There are places where a WildcardPattern is created but it's not used because the string doesn't contains wildcard characters.

This will a perf problem if we will create WildcardPattern _in cycle for large collections_. 
In fact, we see another case - calls of IsMatch() in cycles for real data collection. All cases where we create WildcardPattern and postpone uses is related to small collections like Include parameters as I could find. So it is better to move Init() to constructor and speed up IsMatch()

Reply:
> This will a perf problem if we will create WildcardPattern in cycle for large collections.

The `init` method is very likely to be inlined (I need to check on if multiple return statements can affect inlining, though). If that's the case the difference between eager init and lazy init will just be a check `if (_isMatch == null)`, which doesn't seem to me a perf problem.

Reply:
And I confirmed that JIT inliner can handle multiple returns.

Reply:
Reverted.